Work In Progress - Replace QtCharts with LGPL compliant JKQtPlotter

macumber opened the pull request:
Fixes #836

macumber committed:
First example compiling

macumber committed:
Remove QtCharts from find_package

macumber committed:
Let's see if this will compile on CI

macumber committed:
Initial working version

macumber commented on CMakeLists.txt:
```diff
@@ -658,10 +659,23 @@ if(UNIX)
+                     GIT_REPOSITORY https://github.com/jkriege2/JKQtPlotter.git
+                     GIT_TAG        v4.0.3)
+
+FetchContent_MakeAvailable(JKQTPlotter)
```
@jmarrec we could make a custom conan recipe for this dependency if that is better, but I don't know how to do that

jmarrec replied:
* https://github.com/conan-io/conan-center-index/issues/29109

jmarrec replied:
* https://github.com/conan-io/conan-center-index/issues/29109

macumber commented on src/openstudio_app/CMakeLists.txt:
```diff
@@ -188,7 +188,7 @@ if( APPLE )
 elseif( UNIX )
 
-  foreach(LIB ${QT_LIBS} ${QT_WEB_LIBS})
+  foreach(LIB ${QT_LIBS} ${QT_WEB_LIBS} ${JKQT_LIBS})
```
@jmarrec this is not working on Unix but I'm not sure why

jmarrec replied:
This isn't even built when you are configuring (`_deps/jkqtplotter6-build/output/` is empty).

Also, I don't think we want to be doing this anyways.

We should link to the JKQTPlotter::JKQTPlotter target.

And I think it handles qtdeploy... Or we should just build it statically.

macumber commented:
Seem to be hitting https://github.com/jkriege2/JKQtPlotter/issues/80 on Mac at least

jmarrec committed:
Remove QtCharts6 from AboutBox

jmarrec committed:
Build JKQPLotter statically (8MB)

jmarrec committed:
Add MSVC options for JKQTPLotter per the example

https://github.com/jkriege2/JKQtPlotter/tree/master/examples/cmake_fetchcontent_example

macumber commented:
I'm good to abandon this for now if you want @jmarrec

macumber commented:
Given https://github.com/openstudiocoalition/OpenStudioApplication/pull/859, QtCharts is too nice to get rid of.  

This is addressed by simply listing the licenses in https://github.com/openstudiocoalition/OpenStudioApplication/pull/842
